pkg/hive/job/job.go

```diff
@@ -248,6 +263,7 @@ func (jos *jobOneShot) start(ctx context.Context, wg *sync.WaitGroup, options op
 
 	if jos.fn == nil {
```

Comment:
side-note: can we eliminate this case in the OneShot constructor by making it panic? (likewise for other `fn == nil` checks like timer)